wallet/wallet.c

```diff
@@ -1602,6 +1602,40 @@ void wallet_state_change_add(struct wallet *w,
 	db_exec_prepared_v2(take(stmt));
 }
 
+struct state_change_entry *wallet_state_change_get(struct wallet *w,
+						   const tal_t *ctx,
+						   u64 channel_id)
+{
+	struct db_stmt *stmt;
+	size_t count = 0;
+	struct state_change_entry *res = tal_arr(ctx,
+						 struct state_change_entry, 0);
+	stmt = db_prepare_v2(
+	    w->db, SQL("SELECT"
+		       " timestamp,"
+		       " old_state,"
+		       " new_state,"
+		       " cause,"
+		       " message "
+		       "FROM state_changes "
+		       "WHERE channel_id = ? "
+		       "ORDER BY timestamp ASC;"));
```

Comment:
Good catch :+1: This would not have been necessary on sqlite3 since it maintains insertion order, but it is very much required by postgres.